projects/collection-editor-library/src/lib/components/editor/editor.component.ts

```diff
@@ -253,6 +254,46 @@ export class EditorComponent implements OnInit, OnDestroy, AfterViewInit {
   ));
   }
 
+  sethierarchyConfig(primaryCatConfig) {
+    let hierarchyConfig;
+    if (_.get(primaryCatConfig, 'result.objectCategoryDefinition.objectMetadata.config')) {
+      hierarchyConfig = _.get(primaryCatConfig, 'result.objectCategoryDefinition.objectMetadata.config.sourcingSettings.collection');
+      if (!_.isEmpty(hierarchyConfig.children)) {
+        hierarchyConfig.children = this.getHierarchyChildrenConfig(hierarchyConfig.children);
+      }
+      if (!_.isEmpty(hierarchyConfig.hierarchy)) {
+        _.forEach(hierarchyConfig.hierarchy, (hierarchyValue) => {
+          if (_.get(hierarchyValue, 'children')) {
+            hierarchyConfig['children'] = this.getHierarchyChildrenConfig(_.get(hierarchyValue, 'children'));
+          }
+        });
+      }
+    }
+    this.editorConfig.config = _.assign(this.editorConfig.config, hierarchyConfig);
+  }
+
+  getHierarchyChildrenConfig(childrenData) {
+    _.forEach(childrenData, (value, key) => {
+      if (_.isEmpty(value)) {
+        switch (key) {
+          case 'Question':
```

Comment:
@vaivk369 this logic need to update. Lets have a call

Reply:
Done